pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -417,3 +440,37 @@ func dial(unixSocketPath string, timeout time.Duration) (registerapi.Registratio
 
 	return registerapi.NewRegistrationClient(c), c, nil
 }
+
+func (w *Watcher) containsBlacklistedDir(path string) bool {
+	if w.foundInDeprecatedDir(path) {
+		for _, blacklistSubdir := range deprecatedDirSubpathBlacklist {
+			if i := strings.Index(path, blacklistSubdir); i > -1 {
```

Comment:
was the `kubernetes.io` directory we wanted to avoid because of https://github.com/kubernetes/kubernetes/issues/69015 a direct child of the old plugins dir path? from what I could see in the issue description, it seemed to be. Is it possible to simplify this to something resembling `strings.HasPrefix(path, w.deprecatedPath+"/kubernetes.io/")`? It currently is looking for the blacklisted segment at any point in the directory tree, which is a bit unexpected.

Reply:
It is a direct child of `plugins/` will simplify.